File: scratch/backup_gallery.py
# -*- coding: utf-8 -*-
import os
import re
import requests

# Skript pro lokální zálohu všech fotek z databáze Supabase na váš pevný disk (Python verze)
# Spuštění: python3 scratch/backup_gallery.py

SUPABASE_URL = os.environ.get('SUPABASE_URL', 'https://dxlyjugmeucevosmhage.supabase.co')
SUPABASE_KEY = os.environ.get('SUPABASE_KEY', 'sb_publishable_CR-YuABHB1SvPK6b6sz-WQ_Q6y_8iKx')

# Složka pro uložení fotek (vytvoří se v rootu projektu)
script_dir = os.path.dirname(os.path.abspath(__file__))
BACKUP_DIR = os.path.abspath(os.path.join(script_dir, '..', 'backup_photos'))

# Vytvoření složky, pokud neexistuje
if not os.path.exists(BACKUP_DIR):
    os.makedirs(BACKUP_DIR)

def download_image(url, local_file_path):
    try:
        response = requests.get(url, timeout=30)
        if response.status_code != 200:
            raise Exception(f"Status {response.status_code}")
        with open(local_file_path, 'wb') as f:
            f.write(response.content)
        return True
    except Exception as e:
        print(f"  ❌ Selhalo stahování fotky z {url}: {str(e)}")
        return False

def main():
    print("🚀 Spouštím lokální zálohování galerie fotek ze Supabase...")
    print(f"📂 Fotky se uloží do: {BACKUP_DIR}\n")

    try:
        # 1. Načtení všech záznamů, které mají fotku
        query_url = f"{SUPABASE_URL}/rest/v1/pickups?photo_url=not.is.null&select=id,photo_url,nickname,count,created_at"
        headers = {
            'apikey': SUPABASE_KEY,
            'Authorization': f"Bearer {SUPABASE_KEY}"
        }
        
        response = requests.get(query_url, headers=headers, timeout=30)
        if response.status_code != 200:
            raise Exception(f"Supabase query failed: Status {response.status_code}")

        pickups = response.json()
        print(f"🔍 Nalezeno celkem {len(pickups)} úlovků s fotografií.")

        if len(pickups) == 0:
            print("✅ Žádné fotky k zálohování.")
            return

        downloaded_count = 0
        skipped_count = 0
        failed_count = 0

        # 2. Procházení a stahování
        for pickup in pickups:
            url = pickup.get('photo_url')
            if not url:
                continue

            # Extrakce přípony souboru z URL (výchozí jpg)
            ext = url.split('.')[-1].split('?')[0] if '.' in url else 'jpg'
            if len(ext) > 4 or not ext.isalnum():
                ext = 'jpg'
            
            # Vyčištění přezdívky pro název souboru
            nick = pickup.get('nickname') or 'anonym'
            clean_nick = re.sub(r'[^a-zA-Z0-9]', '_', nick)
            
            # Naformátování data (např. 2026-05-29)
            created_at = pickup.get('created_at')
            date_str = created_at[:10] if created_at else 'unknown_date'
            
            # Strukturovaný název souboru: DATUM_PREZDIVKA_KUSY_ID.PRIPONA
            filename = f"{date_str}_{clean_nick}_{pickup.get('count', 1)}ks_id{pickup['id']}.{ext}"
            local_file_path = os.path.join(BACKUP_DIR, filename)

            # Pokud soubor už lokálně existuje, přeskočíme ho (šetříme čas a přenosy!)
            if os.path.exists(local_file_path):
                skipped_count += 1
                continue

            print(f"📥 Stahuji: {filename}...")
            success = download_image(url, local_file_path)
            if success:
                downloaded_count += 1
            else:
                failed_count += 1

        print(f"\n==================================================")
        print(f"🎉 ZÁLOHA GALERIE DOKONČENA")
        print(f"==================================================")
        print(f"📥 Staženo nových fotek:   {downloaded_count} ks")
        print(f"⏭️ Již existujících (přeskočeno): {skipped_count} ks")
        print(f"❌ Selhalo stahování:      {failed_count} ks")
        print(f"📂 Celkový počet fotek v backup_photos: {len(os.listdir(BACKUP_DIR))} ks")
        print(f"==================================================\n")

    except Exception as e:
        print("❌ Kritická chyba zálohovacího skriptu:", str(e))

if __name__ == '__main__':
    main()

File: scratch/backup_gallery.js
// Skript pro lokální zálohu všech fotek z databáze Supabase na váš pevný disk
// Spuštění: node scratch/backup_gallery.js

const fs = require('fs');
const path = require('path');

const SUPABASE_URL = process.env.SUPABASE_URL || 'https://dxlyjugmeucevosmhage.supabase.co';
const SUPABASE_KEY = process.env.SUPABASE_KEY || 'sb_publishable_CR-YuABHB1SvPK6b6sz-WQ_Q6y_8iKx';

// Složka pro uložení fotek (vytvoří se v rootu projektu)
const BACKUP_DIR = path.join(__dirname, '..', 'backup_photos');

// Vytvoření složky, pokud neexistuje
if (!fs.existsSync(BACKUP_DIR)) {
    fs.mkdirSync(BACKUP_DIR, { recursive: true });
}

// Funkce pro bezpečné stažení a uložení jednoho obrázku
async function downloadImage(url, localFilePath) {
    try {
        const response = await fetch(url);
        if (!response.ok) throw new Error(`Status ${response.status}`);
        const buffer = await response.arrayBuffer();
        fs.writeFileSync(localFilePath, Buffer.from(buffer));
        return true;
    } catch (e) {
        console.error(`  ❌ Selhalo stahování fotky z ${url}:`, e.message);
        return false;
    }
}

async function main() {
    console.log("🚀 Spouštím lokální zálohování galerie fotek ze Supabase...");
    console.log(`📂 Fotky se uloží do: ${BACKUP_DIR}\n`);

    try {
        // 1. Načtení všech záznamů, které mají fotku
        const queryUrl = `${SUPABASE_URL}/rest/v1/pickups?photo_url=not.is.null&select=id,photo_url,nickname,count,created_at`;
        const response = await fetch(queryUrl, {
            headers: {
                'apikey': SUPABASE_KEY,
                'Authorization': `Bearer ${SUPABASE_KEY}`
            }
        });

        if (!response.ok) {
            throw new Error(`Supabase query failed: Status ${response.status}`);
        }

        const pickups = await response.json();
        console.log(`🔍 Nalezeno celkem ${pickups.length} úlovků s fotografií.`);

        if (pickups.length === 0) {
            console.log("✅ Žádné fotky k zálohování.");
            return;
        }

        let downloadedCount = 0;
        let skippedCount = 0;
        let failedCount = 0;

        // 2. Procházení a stahování
        for (const pickup of pickups) {
            const url = pickup.photo_url;
            
            // Extrakce přípony souboru z URL (výchozí jpg)
            const ext = url.split('.').pop().split('?')[0] || 'jpg';
            
            // Vyčištění přezdívky pro název souboru
            const cleanNick = (pickup.nickname || 'anonym').replace(/[^a-zA-Z0-9]/g, '_');
            
            // Naformátování data (např. 2026-05-29)
            const dateStr = pickup.created_at ? pickup.created_at.substring(0, 10) : 'unknown_date';
            
            // Strukturovaný název souboru: DATUM_PREZDIVKA_KUSY_ID.PRIPONA
            const filename = `${dateStr}_${cleanNick}_${pickup.count}ks_id${pickup.id}.${ext}`;
            const localFilePath = path.join(BACKUP_DIR, filename);

            // Pokud soubor už lokálně existuje, přeskočíme ho (šetříme čas a přenosy!)
            if (fs.existsSync(localFilePath)) {
                skippedCount++;
                continue;
            }

            console.log(`📥 Stahuji: ${filename}...`);
            const success = await downloadImage(url, localFilePath);
            if (success) {
                downloadedCount++;
            } else {
                failedCount++;
            }
        }

        console.log(`\n==================================================`);
        console.log(`🎉 ZÁLOHA GALERIE DOKONČENA`);
        console.log(`==================================================`);
        console.log(`📥 Staženo nových fotek:   ${downloadedCount} ks`);
        console.log(`⏭️ Již existujících (přeskočeno): ${skippedCount} ks`);
        console.log(`❌ Selhalo stahování:      ${failedCount} ks`);
        console.log(`📂 Celkový počet fotek v backup_photos: ${fs.readdirSync(BACKUP_DIR).length} ks`);
        console.log(`==================================================\n`);

    } catch (e) {
        console.error("❌ Kritická chyba zálohovacího skriptu:", e);
    }
}

main();
